test/Service/ServiceListenerFactoryTest.php

```diff
@@ -23,25 +24,23 @@ public function setUp()
         $this->factory  = new ServiceListenerFactory();
     }
 
-    /**
-     * @expectedException        Laminas\ServiceManager\Exception\ServiceNotCreatedException
-     * @expectedExceptionMessage The value of service_listener_options must be an array, string given.
-     */
     public function testInvalidOptionType()
     {
+        $this->expectException(ServiceNotCreatedException::class);
+        $this->expectExceptionMessage('The value of service_listener_options must be an array, string given.');
```

Comment:
these lines should be just before call, as it is now always clear where the exception is thrown

here, and all places where expectException has been added below.

Reply:
It was automated fix by phpstorm, I think.
I do not worry about that too much at this point: I intend to do a massive tests refactoring following changes to be introduced across the board for the next major and in order to improve tests quality.

I'll push update shortly